GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -74,43 +79,83 @@ gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
   bitmapFontResourceName,
   texture
 ) {
+  // Reuse an existing bitmapFont that is already in memory and already installed
+  for (var i = 0; i < this._loadedFontsData.length; ++i) {
+    let loadedFontData = this._loadedFontsData[i];
+    if (
+      loadedFontData.fontResourceFileName === bitmapFontResourceName &&
+      PIXI.BitmapFont.available[loadedFontData.fontName]
+    ) {
+      return PIXI.BitmapFont.available[loadedFontData.fontName];
+    }
+  }
+
   let bitmapFontResource = this._resources.find(function (resource) {
     return (
       resource.kind === 'bitmapFont' && resource.name === bitmapFontResourceName
     );
   });
 
   if (!bitmapFontResource) {
-    return Promise.reject(
-      new Error(
-        'Can\'t find resource with name: "' +
-          bitmapFontResourceName +
-          '" (or is not a bitmapFont file).'
-      )
+    console.warn(
+      'Can\'t find resource with name: "' +
+        bitmapFontResourceName +
+        '" (or is not a bitmapFont file).'
     );
-  }
 
-  let response = await fetch(bitmapFontResource.file);
-  if (!response.ok) {
-    return Promise.reject(
-      new Error(
-        "Can't fetch the bitmap font file, error: " +
-          response.status +
-          '(' +
-          response.statusText +
-          ')'
-      )
-    );
+    if (!PIXI.BitmapFont.available['Arial-20-#ffffff-bitmapFont']) {
+      let bitmapFontStyle = new PIXI.TextStyle();
+      bitmapFontStyle.fontFamily = 'Arial';
+      bitmapFontStyle.fontSize = 20;
+      bitmapFontStyle.align = 'left';
+      bitmapFontStyle.wordWrap = true;
+      bitmapFontStyle.fill = '#ffffff';
+
+      const defaultSlugFontName =
+        bitmapFontStyle.fontFamily +
+        '-' +
+        bitmapFontStyle.fontSize +
+        '-' +
+        bitmapFontStyle.fill +
+        '-bitmapFont';
+
+      //Generate default bitmap font
+      PIXI.BitmapFont.from(defaultSlugFontName, bitmapFontStyle, {
+        chars: [
+          [' ', '~'], // All the printable ASCII characters
+        ],
+      });
+      PIXI.BitmapFont.available['Arial-20-#ffffff-bitmapFont'].font =
+        'Arial-20-#ffffff-bitmapFont';
+
+      console.warn('The default bitmap font Arial will be used.');
+    }
+
+    return PIXI.BitmapFont.available['Arial-20-#ffffff-bitmapFont']; // TODO retourné une font par default, le slug Arial par default ou alors une bitmapFont include par default dans GD?
   }
 
-  let fontData = await response.text();
-  let bitmapFont = PIXI.BitmapFont.install(fontData, texture);
-  return Promise.resolve(bitmapFont.font);
+  for (var i = 0; i < this._loadedFontsData.length; ++i) {
+    if (
+      this._loadedFontsData[i].fontResourceFileName === bitmapFontResourceName
+    ) {
+      let fontData = this._loadedFontsData[i].fontData;
+      let bitmapFont = PIXI.BitmapFont.install(fontData, texture);
+
+      this.setFontUsed(bitmapFont.font);
+
+      this._loadedFontsData[i].fontName = bitmapFont.font;
```

Comment:
Could you remove entirely `fontName` and just use the `bitmapFontResourceName`? Resource names are unique.
You might not need to store this name twice (but I'm not sure)

Reply:
An export made with bmFont can export a name file different than the font used inside.

![image](https://user-images.githubusercontent.com/1670670/106812702-526a4a80-6670-11eb-932a-358488c9d261.png)

`bitmapFontResourceName` come from input file can be different than `fontName` from the inside the file.

`bitmapFont.font`